src/chapters/api/documentation.typ

```diff
@@ -0,0 +1,74 @@
+#import "/src/util.typ": *
+#import mantys: *
+
+= Documentation <sec:api:docs>
+Documentaiton comemnts are important for consumers of the API as well as new contributors to your package or project to understand the usage and purpose of an item.
+An item may be a state variable, a counter, a function, a reusable piece of regular content, or anything that can be bound using a `let` statement.
+
+#wbox[
+  For the exact syntax used for documentation comments, refer to @sec:style:docs.
+]
+
+For functions a public API should always document all public arguments, their usage and purpose as well as constraints on their values.
+Internal arguments may be ommited, if they are not meant to be exposed to a user, but should still be documented for contributors in the source or a contribution document.
+Depending on the doc aprser in use, named arguments may need their defaults explicitly stated and should have if the doc parser can't parse them automatically.
+If functions require or return contextual values, this should also be communicated clearly.
+Functions should also document their return type, even if it is general, such as `any`.
+Should a function return different types depending on it's inputs or the state of the document, it must be documented when this may happen or at least which types to expect and check for at the call site.
+
+Values like states and counters, if exposed, should document their invariants, such as the allowed types for states or the expected depth range of a counter for example.
+The type of the value itself should likewise be annotatd using the function return type syntax.
+
+#do-dont[
+  ```typst
+  // - well documented invariants
+  // - most doc parsers will identify named defaults, we omit them for brevity
+  //   in this example
+
+  /// Does things.
+  ///
+  /// This function is contextual if `arg` == `other`.
```

Comment:
It worth mentioning that the named items can be referenced by name in "inline raw" syntax.

Reply:
I don't think there's any established syntax there yet, I use this myself in doc comments, we either allow some form of `@@arg`-like syntax, or don't impose any syntax for it at all.